jenkins/scripts/dynamic_worker_workflow/e2e_tests.sh

```diff
@@ -39,24 +39,33 @@ if [[ "${CAPM3_VERSION}" == "v1alpha5" ]]; then
     export KUBERNETES_VERSION="v1.23.8"
     export KUBECTL_SHA256="${KUBECTL_SHA256:-4685bfcf732260f72fce58379e812e091557ef1dfc1bc8084226c7891dd6028f}"
 fi
-if  [[ -z "${KUBERNETES_VERSION:-}" ]]; then
+
+if [[ "${NUM_NODES}" == "null" ]]; then
+    unset NUM_NODES
+fi
+
+if [[ "${GINKGO_FOCUS}" == "null" ]]; then
+    unset GINKGO_FOCUS
+fi
+
+if [[ "${GINKGO_SKIP}" == "null" ]]; then
     unset GINKGO_SKIP
 fi
 
-if  [[ -z "${KUBECTL_SHA256:-}" ]]; then
-    unset KUBECTL_SHA256
+if [[ "${EPHEMERAL_TEST}" == "null" ]]; then
+    unset EPHEMERAL_TEST
 fi
 
-if  [[ -z "${KUBERNETES_VERSION_UPGRADE_FROM:-}" ]]; then
+if [[ "${KUBERNETES_VERSION_UPGRADE_FROM}" == "null" ]]; then
     unset KUBERNETES_VERSION_UPGRADE_FROM
 fi
 
-if  [[ -z "${KUBERNETES_VERSION_UPGRADE_TO:-}" ]]; then
+if [[ "${KUBERNETES_VERSION_UPGRADE_TO}" == "null" ]]; then
     unset KUBERNETES_VERSION_UPGRADE_TO
 fi
 
-if  [[ -z "${EPHEMERAL_TEST:-}" ]]; then
-    unset EPHEMERAL_TEST
+if [[ "${KUBECTL_SHA256}" == "null" ]]; then
```

Comment:
An idea could be to configure all environment variables always as a parameter in Jenkins job then we could have the default as empty string or null. In my opinion that would be a bit more explicit and work a bit as a documentation as currently it was somewhat confusing at least to me from where this null came. What do you think about that?

Reply:
That is the issue of null value in groove and in shell scripts. When value is not defined in pipeline script it is set to null, and that null is converted as a string null into shell. I am not sure if we can fix that on jenkins side

Reply:
I meant defining them in Jenkins parameters and I think that should be able to fix it with a default value ("" or null) there see below, but actually I like @tuminoid idea more it is simpler. Maybe make it a function and the function can document why it is done?

![image](https://github.com/metal3-io/project-infra/assets/55130152/4f2696aa-6c40-402c-9865-fc3ada42e19f)